homeassistant/components/homepluscontrol/switch.py

```diff
@@ -0,0 +1,216 @@
+    )
+
+    # Add the coordinator to the domain's data in HA
+    hass.data[DOMAIN][config_entry.entry_id + "_coordinator"] = coordinator
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+
+async def view_data(hass):
+    """Debug relevant Hass objects."""
+    device_reg = await hass.helpers.device_registry.async_get_registry()
+    entity_reg = await hass.helpers.entity_registry.async_get_registry()
+    for ent in entity_reg.entities:
+        _LOGGER.debug("Entity registry entity: %s", str(ent))
+    for dev in device_reg.devices:
+        _LOGGER.debug("Device registry device: %s", str(dev))
+
+
+class HomeControlSwitchEntity(CoordinatorEntity, SwitchEntity):
+    """Entity that represents a Legrand Home+ Control switch.
+
+    It extends the HomeAssistant-provided classes of the CoordinatorEntity and the SwitchEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+
+    The SwitchEntity class provides the functionality of a ToggleEntity and additional power
+    consumption methods and state attributes.
+    """
+
+    def __init__(self, coordinator, idx):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.idx = idx
+
+    @property
+    def name(self):
+        """Name of the device."""
+        return self.coordinator.data[self.idx].name
+
+    @property
+    def unique_id(self):
+        """ID (unique) of the device."""
+        return self.idx
+
+    @property
+    def device_info(self):
+        """Device information."""
+        return {
+            "identifiers": {
+                # Unique identifiers within the domain
+                (DOMAIN, self.unique_id)
+            },
+            "name": self.name,
+            "manufacturer": "Legrand",
+            "model": HW_TYPE.get(self.coordinator.data[self.idx].hw_type, "Unknown"),
+            "sw_version": self.coordinator.data[self.idx].fw,
+        }
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        if self.coordinator.data[self.idx].device == "plug":
+            return DEVICE_CLASS_OUTLET
+        return DEVICE_CLASS_SWITCH
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Logger of entity."""
+        return logging.getLogger(__name__)
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available.
+
+        This is the case when the coordinator is able to update the data successfully
+        AND the switch entity is reachable.
+
+        This method overrides the one of the CoordinatorEntity
+        """
+        return (
+            self.coordinator.last_update_success
+            and self.coordinator.data[self.idx].reachable
+        )
+
+    @property
+    def is_on(self):
+        """Return entity state."""
+        self.logger.debug(
```

Comment:
We don't allow side effects like logging in entity state properties.

Reply:
Removed the logging actions from this part of the code.